### ISSUE TITLE
Remove DockerHub "pull credentials" from Kubernetes clusters

### ISSUE THREAD

**Issue by dduportal** [jenkins-infra/helpdesk#3577]
### Service(s)

Other

### Summary

TL;DR; rollback https://github.com/jenkins-infra/helpdesk/issues/2830.

- As per https://github.com/jenkins-infra/helpdesk/issues/3452, all Docker images in `jenkinsciinfra/*` and `jenkins/*` are not API-rate limited: as such no more pull credentials are required
- Datadog images are now hosted on gcr.io (https://github.com/DataDog/helm-charts/blob/7b3e88ee5e4cabebc084ebc7b29c18bc51a930d4/charts/datadog/values.yaml#L23-L27) so no need for the pull credntial anymore

Steps:

- In kubernetes-management, remove any mention of the `docker-registry-secrets` chart (such as https://github.com/jenkins-infra/kubernetes-management/blob/main/clusters/cik8s.yaml#L20-L29).
  - Might need to remove mentions of the pull crednetial in jenkins controllers setups AND datadog agent values if any
  - Wait 24hours after merging this change before removing manually the resources in the clusters
- Remove the charts from jenkins-infra/helm-charts
- Remove the token from DockerHub 
- Remove the tokens from chart-secrets

**Comment by dduportal** [comment]
- Chart installation removed in https://github.com/jenkins-infra/kubernetes-management/commit/0a13913fc6104cd4260664ece394f5cafb0308b2
- Chart deprecated in https://github.com/jenkins-infra/helpdesk/issues/4460
- SOPS secrets removed in https://github.com/jenkins-infra/charts-secrets/commit/f3d8088e2bf63862c137e27d7dec30ba9b276f38, https://github.com/jenkins-infra/charts-secrets/commit/38947fe5b5f844e90b89026ad757180126e647ac

=> no token to remove on DockerHub (as still used by the transparent registry mirrors)
